src/words/environment/WordsEnvironment.java

```diff
@@ -10,13 +12,21 @@
 
 public class WordsEnvironment {
 	private HashMap<String, WordsClass> classes;
-	private HashMap<String, WordsObject> objects;
+	/* gettableObjects is a list of locally scoped symbol tables with the most local 
+	 * scope first in the list, and global scope always being last in the list.
+	 */
+	private LinkedList<HashMap<String, WordsObject>> gettableObjects;
+	/* ungettableObjects contains the objects that have gone out of scope, but should still appear
+	 * on the board, and are still used by action listeners */
+	private HashSet<WordsObject> ungettableObjects;
```

Comment:
Is the advantage of using a HashSet here that, later on when we need to remove an object from all our data structures that reference it, we can do so directly?  As it is now, it seems we don't use the fact that ungettableObjects is a HashSet versus some other structure.